src/step.rs

```diff
@@ -191,3 +191,64 @@ impl Ord for Step {
     }
 }
 
+macro_rules! impl_from {
+    ($TargetType:ty) => {
+        impl From<$TargetType> for Step {
+            fn from(val: $TargetType) -> Self {
+                Step(val as f32)
+            }
+        }
+    };
+}
+
+macro_rules! impl_all_from {
+    (
+        $($x:ty),+
+    ) => {
+        $(
+            impl_from!($x);
+        )*
+    };
+}
+
+impl_all_from!(u8, u16, u32, u64, u128, i8, i16, i32, i64, i128);
+
+#[cfg(test)]
+mod tests {
+    use super::Step;
+
+    macro_rules! t {
+        (
+            $($x:ty),+
+        ) => {
+            fn from(val: f32) {
+                $(
+                    assert_eq!(Step::from(val as $x), Step(val));
+                )*
+            }
+
+            #[test]
+            fn test_from_integer() {
+                from(0.0);
+                from(60.0);
+                from(127.0);
+            }
+
+            fn into(val: f32) {
+                $(
+                    let actual: Step = (val as $x).into();
```

Comment:
The intermediate variable is necessary because otherwise the Rust compiler is not able to infer the correct type to convert `into()`.